### PR TITLE
Components: remove `color()` utility function, use values directly

### PR DESCRIPTION
**Pull request by ciampo**
## Description

Remove the `color()` utility function, and instead access the color values directly.

This is similar to what has been done in #31646 for the `config()` function, and is part of what highlighted in #30503

## How has this been tested?

- [ ] Make sure that the project builds correctly
- [ ] Make sure that the affected components look and behave the same as on `trunk` (e.g. by using [Storybook](https://github.com/WordPress/gutenberg/blob/trunk/storybook/README.md))

## Screenshots 

N/A

## Types of changes

Refactor / cleanup

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] N/A I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] N/A I've included developer documentation if appropriate. 
- [ ] N/A I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 ### Internal
 
 -   `Flex`, `FlexBlock`, and `FlexItem` components have been re-written from the ground up ([#31297](https://github.com/WordPress/gutenberg/pull/31297)).
+-   `color()` utility function has been removed, in favour of accessing color values directly ([#31661](https://github.com/WordPress/gutenberg/pull/31661))
```
Comment (tyxla) [inline packages/components/CHANGELOG.md]: Nit: we're mixing American and British English when using `color` (US) but `favour` (UK). I guess we should stick to the American version as it's the one used in web terms.

```suggestion
-   `color()` utility function has been removed, in favor of accessing color values directly ([#31661](https://github.com/WordPress/gutenberg/pull/31661))
```

**Review by tyxla — APPROVED** [review]
Code looks good and this tests well for me in Storybook 👍

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md]: Well spotted! I normally write in British English, and so this happens more often than I'd like 😅

Actually, in light of @diegohaz 's [comment on a very similar PR](https://github.com/WordPress/gutenberg/pull/31646#discussion_r630395238), should I actually just delete this changelog entry?

Comment (tyxla) [reply packages/components/CHANGELOG.md]: Yup, agree that changes in non-exposed internals don't need to be documented here 👍

Comment (ciampo) [reply packages/components/CHANGELOG.md]: Done in https://github.com/WordPress/gutenberg/pull/31661/commits/dd4b976853827c3311f802232d9800f7103b7870

**Comment by ciampo** [comment]
> Code looks good and this tests well for me in Storybook 👍

Thank you for your review, @tyxla ! 

Since I don't have repo privileges yet, would you mind merging this PR? Thank you!

**Comment by tyxla** [comment]
> Since I don't have repo privileges yet, would you mind merging this PR? Thank you!

Sure thing! Thanks for your work here 🙌

**Pull request merged**